### ISSUE TITLE
[Enhancement] Allow monitoring volume and mute status of audio input devices 

### ISSUE THREAD

**Issue by Adaephon-GH** [i3/i3status#241]
At the moment it looks like it is only possible to monitor audio output devices. (At least for me any attempts to add an ALSA capture device or PulseAudio source failed)

 It would be great if it would also be possible to monitor input devices.

**Comment by reitzig** [comment]
Definitely! Accidentally muted or tuned down microphones are a constant point of confusion in video conferences. It would be great to have "volumne" indicators (gain) for inputs in the bar!

**Comment by Stunkymonkey** [comment]
I started implementing this.

**Comment by eplanet** [comment]
@Stunkymonkey any update on this?

**Comment by Stunkymonkey** [comment]
the alsa part ist working, but the pulse part not

**Comment by nidrissi** [comment]
@Airblader The fix was only for ALSA, it doesn't work with PulseAudio.

**Comment by Airblader** [comment]
Yes, thanks!

**Comment by qguv** [comment]
This would be excellent to have; has there been any work on this for PulseAudio since 2017?

**Comment by grmat** [comment]
This has the `help wanted` label, but there's an open PR at #345 with no further discussion. Could you elaborate on what's missing/todo?

**Comment by orestisfl** [comment]
That PR doesn't link to this issue so I think I didn't notice it

**Comment by grmat** [comment]
Alright. FWIW, I fixed an issue that I personally had with the current ALSA implementation in #423, but I'd be willing to work on Pulse integration as well if necessary.

**Comment by nitish-predera** [comment]
I wish this feature gets to breathe air.

**Comment by ulfgebhardt** [comment]
Can we have this please <3

**Comment by n-st** [comment]
As a workaround, you can use a wrapper script around i3status that queries the input device status on its own and adds it to the status line — something like this: https://gist.github.com/n-st/c91fb5a423c539f3b2cb1013384c8af2
The script is very simplistic, but will detect the current default source device and replace the "MIC placeholder" status item with a red "MIC" or a white "MUT" depending on the current mute status. Feel free to modify it for your desired format (example code for a volume meter is included as well).
